plugins/woocommerce/src/Blocks/BlockTypes/BlockifiedProductDetails.php

```diff
@@ -263,13 +301,111 @@ private function has_accordion( $parsed_block ) {
 	}
 
 	/**
-	 * Get the frontend script handle for this block type.
+	 * Validate hooked blocks data. Remove duplicated entries with the same title
+	 * and invalid entries with invalid content. Log errors to the WC logger.
 	 *
-	 * @see $this->register_block_type()
-	 * @param string $key Data to get, or default to everything.
-	 * @return array|string|null
+	 * @param array $hooked_blocks { Hooked blocks data.
+	 *   @type string $title Title of the hooked block.
+	 *   @type string $content Content of the hooked block, as block markup.
+	 * }
+	 *
+	 * @return array Validated hooked blocks.
 	 */
-	protected function get_block_type_script( $key = null ) {
-		return null;
+	private function validate_hooked_blocks( $hooked_blocks ) {
+		$logger                  = wc_get_logger();
+		$validated_hooked_blocks = [];
+
+		foreach ( $hooked_blocks as $block ) {
+			$invalid = ! is_array( $block ) ||
+			! isset( $block['title'] ) ||
+			! isset( $block['content'] ) ||
+			! is_string( $block['title'] ) ||
+			! is_string( $block['content'] );
+
+			$parsed_content = parse_blocks( $block['content'] );
+
+			foreach ( $parsed_content as $content_block ) {
+				if ( ! isset( $content_block['blockName'] ) ) {
+					$invalid = true;
+					break;
+				}
+			}
+
+			if ( $invalid ) {
+				$logger->error( 'Invalid hooked block data. Expected array with `title` and `content` keys with string values. Content must be valid block markup.', $block );
+				continue;
+			}
+
+			if ( array_filter(
+				$validated_hooked_blocks,
+				function ( $hooked_block ) use ( $block ) {
+					return $hooked_block['title'] === $block['title'];
+				}
+			) ) {
+				$logger->error( 'Duplicate hooked block data. Hooked block with title `' . $block['title'] . '` is already registered.', $block );
+				continue;
+			}
+
+			$validated_hooked_blocks[] = $block;
+		}
+
+		return $validated_hooked_blocks;
+	}
+
+	/**
+	 * Register a product details item using Block Hooks API.
+	 *
+	 * @param string $title The title of the item.
+	 * @param string $content The content of the item.
+	 * @return void
+	 */
+	private function register_hooked_blocks( $title, $content ) {
```

Comment:
Nitpick: As it stands now, the function only registers one hooked block, so its name should probably reflect that:

```suggestion
	private function register_hooked_block( $title, $content ) {
```